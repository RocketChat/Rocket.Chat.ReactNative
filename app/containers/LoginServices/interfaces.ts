import { ReactElement } from 'react';

import { IServices } from '../../selectors/login';
import { TIconsName } from '../CustomIcon';

type TAuthType = 'oauth' | 'oauth_custom' | 'saml' | 'cas' | 'apple';

type TServiceName = 'facebook' | 'github' | 'gitlab' | 'google' | 'linkedin' | 'meteor-developer' | 'twitter' | 'wordpress';
export interface IOpenOAuth {
	url: string;
	ssoToken?: string;
	authType?: TAuthType;
}

export interface IItemService {
	_id: string;
	name: TServiceName;
	service: string;
	authType: TAuthType;
	buttonColor: string;
	buttonLabelColor: string;
	clientConfig: { provider: string };
	serverURL: string;
	authorizePath: string;
	clientId: string;
	scope: string;
}

export interface IServiceLogin {
	service: IItemService;
	server: string;
	urlOption?: string;
}

export interface IOauthProvider {
	[key: string]: ({ service, server }: IServiceLogin) => void;
	facebook: ({ service, server }: IServiceLogin) => void;
	github: ({ service, server }: IServiceLogin) => void;
	gitlab: ({ service, server }: IServiceLogin) => void;
	google: ({ service, server }: IServiceLogin) => void;
	linkedin: ({ service, server }: IServiceLogin) => void;
	'meteor-developer': ({ service, server }: IServiceLogin) => void;
	twitter: ({ service, server }: IServiceLogin) => void;
	wordpress: ({ service, server }: IServiceLogin) => void;
}

export interface IServiceList {
	services: IServices;
	CAS_enabled: boolean;
	CAS_login_url: string;
	Gitlab_URL: string;
	server: string;
}

export interface IServicesSeparator {
	services: IServices;
	separator: boolean;
	collapsed: boolean;
	onPress(): void;
}

export interface IButtonService {
	name: string;
	authType: TAuthType;
	onPress: () => void;
	backgroundColor: string;
	buttonText: ReactElement;
	icon: TIconsName;
}
