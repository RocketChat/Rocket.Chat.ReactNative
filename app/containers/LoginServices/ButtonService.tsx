import React from 'react';
import { Text, View } from 'react-native';

import { useTheme } from '../../theme';
import Touch from '../Touch';
import { CustomIcon } from '../CustomIcon';
import { IButtonService } from './interfaces';
import styles from './styles';

const ButtonService = ({ name, authType, onPress, backgroundColor, buttonText, icon }: IButtonService) => {
	const { colors } = useTheme();

	return (
		<Touch
			key={name}
			onPress={onPress}
			style={[styles.serviceButton, { backgroundColor }]}
			activeOpacity={0.5}
			underlayColor={colors.buttonText}
		>
			<View style={styles.serviceButtonContainer}>
				{authType === 'oauth' || authType === 'apple' ? (
					<CustomIcon name={icon} size={24} color={colors.titleText} style={styles.serviceIcon} />
				) : null}
				<Text style={[styles.serviceText, { color: colors.titleText }]}>{buttonText}</Text>
			</View>
		</Touch>
	);
};

export default ButtonService;
