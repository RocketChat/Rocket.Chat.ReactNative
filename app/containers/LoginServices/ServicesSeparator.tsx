import React from 'react';

import Button from '../Button';
import OrSeparator from '../OrSeparator';
import { useTheme } from '../../theme';
import styles from './styles';
import I18n from '../../i18n';
import { IServicesSeparator } from './interfaces';

const ServicesSeparator = ({ services, separator, collapsed, onPress }: IServicesSeparator) => {
	const { colors, theme } = useTheme();

	const { length } = Object.values(services);

	if (length > 3 && separator) {
		return (
			<>
				<Button
					title={collapsed ? I18n.t('Onboarding_more_options') : I18n.t('Onboarding_less_options')}
					type='secondary'
					onPress={onPress}
					style={styles.options}
					color={colors.actionTintColor}
				/>
				<OrSeparator theme={theme} />
			</>
		);
	}
	if (length > 0 && separator) {
		return <OrSeparator theme={theme} />;
	}
	return null;
};

export default ServicesSeparator;
