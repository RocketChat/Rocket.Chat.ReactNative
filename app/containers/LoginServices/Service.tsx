import React, { useRef } from 'react';
import { Text } from 'react-native';

import { useTheme } from '../../theme';
import I18n from '../../i18n';
import { TIconsName } from '../CustomIcon';
import { IItemService, IOauthProvider } from './interfaces';
import styles from './styles';
import * as ServiceLogin from './serviceLogin';
import ButtonService from './ButtonService';

const Service = React.memo(
	({
		CAS_enabled,
		CAS_login_url,
		Gitlab_URL,
		server,
		service
	}: {
		service: IItemService;
		server: string;
		Gitlab_URL: string;
		CAS_enabled: boolean;
		CAS_login_url: string;
		storiesTestOnPress?: () => void;
	}) => {
		const { colors } = useTheme();
		const onPress = useRef<any>();
		const buttonText = useRef<React.ReactElement>();
		const modifiedName = useRef<string>();

		const { name } = service;
		modifiedName.current = name === 'meteor-developer' ? 'meteor' : name;
		const icon = `${modifiedName.current}-monochromatic` as TIconsName;
		const isSaml = service.service === 'saml';

		const getSocialOauthProvider = (name: string) => {
			const oauthProviders: IOauthProvider = {
				facebook: () => ServiceLogin.onPressFacebook({ service, server }),
				github: () => ServiceLogin.onPressGithub({ service, server }),
				gitlab: () => ServiceLogin.onPressGitlab({ service, server, urlOption: Gitlab_URL }),
				google: () => ServiceLogin.onPressGoogle({ service, server }),
				linkedin: () => ServiceLogin.onPressLinkedin({ service, server }),
				'meteor-developer': () => ServiceLogin.onPressMeteor({ service, server }),
				twitter: () => ServiceLogin.onPressTwitter({ service, server }),
				wordpress: () => ServiceLogin.onPressWordpress({ service, server })
			};
			return oauthProviders[name];
		};

		switch (service.authType) {
			case 'oauth': {
				onPress.current = getSocialOauthProvider(service.name);
				break;
			}
			case 'oauth_custom': {
				onPress.current = () => ServiceLogin.onPressCustomOAuth({ loginService: service, server });
				break;
			}
			case 'saml': {
				onPress.current = () => ServiceLogin.onPressSaml({ loginService: service, server });
				break;
			}
			case 'cas': {
				onPress.current = () => ServiceLogin.onPressCas({ casLoginUrl: CAS_login_url, server });
				break;
			}
			case 'apple': {
				onPress.current = () => ServiceLogin.onPressAppleLogin();
				break;
			}
			default:
				break;
		}

		modifiedName.current = modifiedName.current.charAt(0).toUpperCase() + modifiedName.current.slice(1);
		if (isSaml || (service.service === 'cas' && CAS_enabled)) {
			buttonText.current = (
				<Text style={[styles.serviceName, isSaml && { color: service.buttonLabelColor }]}>{modifiedName.current}</Text>
			);
		} else {
			buttonText.current = (
				<>
					{I18n.t('Continue_with')} <Text style={styles.serviceName}>{modifiedName.current}</Text>
				</>
			);
		}

		const backgroundColor = isSaml && service.buttonColor ? service.buttonColor : colors.chatComponentBackground;

		return (
			<ButtonService
				onPress={onPress.current}
				backgroundColor={backgroundColor}
				buttonText={buttonText.current}
				icon={icon}
				name={service.name}
				authType={service.authType}
			/>
		);
	}
);

export default Service;
