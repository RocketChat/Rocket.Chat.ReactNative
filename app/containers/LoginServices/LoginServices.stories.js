import React from 'react';
import { storiesOf } from '@storybook/react-native';
import { Provider } from 'react-redux';
import { StyleSheet, Text, ScrollView } from 'react-native';

import { store } from '../../../storybook/stories';
import { ThemeContext } from '../../theme';
import { colors } from '../../lib/constants';
import i18n from '../../i18n';
import sharedStyles from '../../views/Styles';
import ServicesSeparator from './ServicesSeparator';
import ButtonService from './ButtonService';

const styles = StyleSheet.create({
	serviceName: {
		...sharedStyles.textSemibold
	}
});

const services = {
	github: {
		_id: 'github',
		name: 'github',
		clientId: 'github-123',
		buttonLabelText: '',
		buttonColor: '',
		buttonLabelColor: '',
		custom: false,
		authType: 'oauth'
	},
	gitlab: {
		_id: 'gitlab',
		name: 'gitlab',
		clientId: 'gitlab-123',
		buttonLabelText: '',
		buttonColor: '',
		buttonLabelColor: '',
		custom: false,
		authType: 'oauth'
	},
	google: {
		_id: 'google',
		name: 'google',
		clientId: 'google-123',
		buttonLabelText: '',
		buttonColor: '',
		buttonLabelColor: '',
		custom: false,
		authType: 'oauth'
	},
	apple: {
		_id: 'apple',
		name: 'apple',
		clientId: 'apple-123',
		buttonLabelText: 'Sign in with Apple',
		buttonColor: '#000',
		buttonLabelColor: '#FFF',
		custom: false,
		authType: 'apple'
	}
};

const theme = 'light';

const stories = storiesOf('Login Services', module)
	.addDecorator(story => <Provider store={store}>{story()}</Provider>)
	.addDecorator(story => <ThemeContext.Provider value={{ theme, colors: colors[theme] }}>{story()}</ThemeContext.Provider>)
	.addDecorator(story => <ScrollView style={sharedStyles.containerScrollView}>{story()}</ScrollView>);

stories.add('ServicesSeparator', () => (
	<>
		<ServicesSeparator collapsed onPressButtonSeparator={() => {}} separator services={services} />
		<ServicesSeparator collapsed={false} onPressButtonSeparator={() => {}} separator services={services} />
	</>
));

stories.add('ServiceList', () => (
	<>
		{Object.values(services).map(service => {
			const icon = `${service.name}-monochromatic`;
			const buttonText = (
				<>
					{i18n.t('Continue_with')} <Text style={styles.serviceName}>{service.name}</Text>
				</>
			);
			return (
				<ButtonService
					key={service._id}
					onPress={() => {}}
					backgroundColor={colors[theme].chatComponentBackground}
					buttonText={buttonText}
					icon={icon}
					name={service.name}
					authType={service.authType}
				/>
			);
		})}
	</>
));
