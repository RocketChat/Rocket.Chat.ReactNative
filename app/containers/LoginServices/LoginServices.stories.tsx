import React from 'react';
import { StyleSheet, Text } from 'react-native';

import { colors } from '../../lib/constants';
import i18n from '../../i18n';
import sharedStyles from '../../views/Styles';
import ServicesSeparator from './ServicesSeparator';
import ButtonService from './ButtonService';
import { IServices } from '../../selectors/login';
import { TIconsName } from '../CustomIcon';

const styles = StyleSheet.create({
	serviceName: {
		...sharedStyles.textSemibold
	}
});

const services = {
	github: {
		_id: 'github',
		name: 'github',
		clientId: 'github-123',
		buttonLabelText: '',
		buttonColor: '',
		buttonLabelColor: '',
		custom: false,
		authType: 'oauth'
	},
	gitlab: {
		_id: 'gitlab',
		name: 'gitlab',
		clientId: 'gitlab-123',
		buttonLabelText: '',
		buttonColor: '',
		buttonLabelColor: '',
		custom: false,
		authType: 'oauth'
	},
	google: {
		_id: 'google',
		name: 'google',
		clientId: 'google-123',
		buttonLabelText: '',
		buttonColor: '',
		buttonLabelColor: '',
		custom: false,
		authType: 'oauth'
	},
	apple: {
		_id: 'apple',
		name: 'apple',
		clientId: 'apple-123',
		buttonLabelText: 'Sign in with Apple',
		buttonColor: '#000',
		buttonLabelColor: '#FFF',
		custom: false,
		authType: 'apple'
	}
} as unknown as IServices;

const theme = 'light';

export default {
	title: 'Login Services'
};

export const Separators = () => (
	<>
		<ServicesSeparator collapsed onPress={() => {}} separator services={services} />
		<ServicesSeparator collapsed={false} onPress={() => {}} separator services={services} />
	</>
);

export const ServiceList = () => (
	<>
		{Object.values(services).map(service => {
			const icon = `${service.name}-monochromatic` as TIconsName;
			const buttonText = (
				<>
					{i18n.t('Continue_with')} <Text style={styles.serviceName}>{service.name}</Text>
				</>
			);
			return (
				<ButtonService
					key={service._id}
					onPress={() => {}}
					backgroundColor={colors[theme].chatComponentBackground}
					buttonText={buttonText}
					icon={icon}
					name={service.name}
					authType={service.authType}
				/>
			);
		})}
	</>
);
