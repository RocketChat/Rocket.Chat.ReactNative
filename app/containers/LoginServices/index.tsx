import React, { useState } from 'react';
import { shallowEqual } from 'react-redux';
import Animated, { Easing, useAnimatedStyle, useSharedValue, withTiming } from 'react-native-reanimated';

import { IServices } from '../../selectors/login';
import { useAppSelector } from '../../lib/hooks';
import { IItemService, IServiceList } from './interfaces';
import { SERVICES_COLLAPSED_HEIGHT, SERVICE_HEIGHT } from './styles';
import ServicesSeparator from './ServicesSeparator';
import Service from './Service';

const ServiceList = ({ services, CAS_enabled, CAS_login_url, Gitlab_URL, server }: IServiceList) => (
	<>
		{Object.values(services).map((service: IItemService) => (
			<Service
				key={service._id}
				CAS_enabled={CAS_enabled}
				CAS_login_url={CAS_login_url}
				Gitlab_URL={Gitlab_URL}
				server={server}
				service={service}
			/>
		))}
	</>
);

const LoginServices = ({ separator }: { separator: boolean }): React.ReactElement => {
	const [collapsed, setCollapsed] = useState(true);

	const { Gitlab_URL, CAS_enabled, CAS_login_url } = useAppSelector(
		state => ({
			Gitlab_URL: state.settings.API_Gitlab_URL as string,
			CAS_enabled: state.settings.CAS_enabled as boolean,
			CAS_login_url: state.settings.CAS_login_url as string
		}),
		shallowEqual
	);
	const server = useAppSelector(state => state.server.server);
	const services = useAppSelector(state => state.login.services as IServices, shallowEqual);
	const { length } = Object.values(services);

	const heightButtons = useSharedValue(SERVICES_COLLAPSED_HEIGHT);

	const animatedStyle = useAnimatedStyle(() => ({
		overflow: 'hidden',
		height: withTiming(heightButtons.value, { duration: 300, easing: Easing.inOut(Easing.quad) })
	}));

	const onPressButtonSeparator = () => {
		heightButtons.value = collapsed ? SERVICE_HEIGHT * length : SERVICES_COLLAPSED_HEIGHT;
		setCollapsed(prevState => !prevState);
	};

	if (length > 3 && separator) {
		return (
			<>
				<Animated.View style={animatedStyle}>
					<ServiceList
						services={services}
						CAS_enabled={CAS_enabled}
						CAS_login_url={CAS_login_url}
						Gitlab_URL={Gitlab_URL}
						server={server}
					/>
				</Animated.View>
				<ServicesSeparator services={services} separator={separator} collapsed={collapsed} onPress={onPressButtonSeparator} />
			</>
		);
	}
	return (
		<>
			<ServiceList
				services={services}
				CAS_enabled={CAS_enabled}
				CAS_login_url={CAS_login_url}
				Gitlab_URL={Gitlab_URL}
				server={server}
			/>
			<ServicesSeparator services={services} separator={separator} collapsed={collapsed} onPress={onPressButtonSeparator} />
		</>
	);
};

export default LoginServices;
