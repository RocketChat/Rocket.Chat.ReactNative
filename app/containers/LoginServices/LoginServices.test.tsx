import { generateSnapshots } from '../../../.rnstorybook/generateSnapshots';
import * as stories from './LoginServices.stories';

generateSnapshots(stories);
