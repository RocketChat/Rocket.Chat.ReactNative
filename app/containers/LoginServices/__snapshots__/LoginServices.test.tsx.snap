// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Story Snapshots: Separators should match snapshot 1`] = `
[
  <View
    accessibilityLabel="More options"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#E4E7EA",
        "borderRadius": 4,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 14,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#6C727A",
            "fontSize": 16,
          },
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontWeight": "500",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      More options
    </Text>
  </View>,
  <View
    accessible={true}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginVertical": 24,
      }
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
            "height": 1,
          },
          {
            "backgroundColor": "#CBCED1",
          },
        ]
      }
    />
    <Text
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 16,
            "fontWeight": "500",
            "marginHorizontal": 12,
            "textAlign": "left",
          },
          {
            "color": "#6C727A",
          },
        ]
      }
    >
      OR
    </Text>
    <View
      style={
        [
          {
            "flex": 1,
            "height": 1,
          },
          {
            "backgroundColor": "#CBCED1",
          },
        ]
      }
    />
  </View>,
  <View
    accessibilityLabel="Less options"
    accessibilityRole="button"
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "backgroundColor": "#E4E7EA",
        "borderRadius": 4,
        "justifyContent": "center",
        "marginBottom": 0,
        "opacity": 1,
        "paddingHorizontal": 16,
        "paddingVertical": 14,
      }
    }
  >
    <Text
      style={
        [
          {
            "color": "#6C727A",
            "fontSize": 16,
          },
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontWeight": "500",
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      Less options
    </Text>
  </View>,
  <View
    accessible={true}
    style={
      {
        "alignItems": "center",
        "flexDirection": "row",
        "marginVertical": 24,
      }
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
            "height": 1,
          },
          {
            "backgroundColor": "#CBCED1",
          },
        ]
      }
    />
    <Text
      style={
        [
          {
            "backgroundColor": "transparent",
            "fontFamily": "Inter",
            "fontSize": 16,
            "fontWeight": "500",
            "marginHorizontal": 12,
            "textAlign": "left",
          },
          {
            "color": "#6C727A",
          },
        ]
      }
    >
      OR
    </Text>
    <View
      style={
        [
          {
            "flex": 1,
            "height": 1,
          },
          {
            "backgroundColor": "#CBCED1",
          },
        ]
      }
    />
  </View>,
]
`;

exports[`Story Snapshots: ServiceList should match snapshot 1`] = `
[
  <RNGestureHandlerButton
    activeOpacity={0.5}
    collapsable={false}
    delayLongPress={600}
    handlerTag={1}
    handlerType="NativeViewGestureHandler"
    innerRef={null}
    onActiveStateChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onPress={[Function]}
    rippleColor="#E4E7EA"
    style={
      [
        {
          "backgroundColor": "#F7F8FA",
          "borderRadius": 4,
          "margin": undefined,
          "marginBottom": 10,
          "marginEnd": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginStart": undefined,
          "marginTop": undefined,
          "marginVertical": undefined,
        },
        {
          "cursor": undefined,
        },
      ]
    }
    underlayColor="#FFFFFF"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderRadius": 4,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessible={true}
      style={{}}
    >
      <View
        accessible={false}
        style={
          {
            "alignItems": "center",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 15,
            "paddingVertical": 14,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#2F343D",
                "fontSize": NaN,
              },
              [
                {
                  "lineHeight": NaN,
                },
                {
                  "left": 15,
                  "position": "absolute",
                },
              ],
              {
                "fontFamily": "custom",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          î¤‡
        </Text>
        <Text
          accessible={false}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Continue with
           
          <Text
            style={
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontWeight": "600",
                "textAlign": "left",
              }
            }
          >
            github
          </Text>
        </Text>
      </View>
    </View>
  </RNGestureHandlerButton>,
  <RNGestureHandlerButton
    activeOpacity={0.5}
    collapsable={false}
    delayLongPress={600}
    handlerTag={2}
    handlerType="NativeViewGestureHandler"
    innerRef={null}
    onActiveStateChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onPress={[Function]}
    rippleColor="#E4E7EA"
    style={
      [
        {
          "backgroundColor": "#F7F8FA",
          "borderRadius": 4,
          "margin": undefined,
          "marginBottom": 10,
          "marginEnd": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginStart": undefined,
          "marginTop": undefined,
          "marginVertical": undefined,
        },
        {
          "cursor": undefined,
        },
      ]
    }
    underlayColor="#FFFFFF"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderRadius": 4,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessible={true}
      style={{}}
    >
      <View
        accessible={false}
        style={
          {
            "alignItems": "center",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 15,
            "paddingVertical": 14,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#2F343D",
                "fontSize": NaN,
              },
              [
                {
                  "lineHeight": NaN,
                },
                {
                  "left": 15,
                  "position": "absolute",
                },
              ],
              {
                "fontFamily": "custom",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          î¤ˆ
        </Text>
        <Text
          accessible={false}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Continue with
           
          <Text
            style={
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontWeight": "600",
                "textAlign": "left",
              }
            }
          >
            gitlab
          </Text>
        </Text>
      </View>
    </View>
  </RNGestureHandlerButton>,
  <RNGestureHandlerButton
    activeOpacity={0.5}
    collapsable={false}
    delayLongPress={600}
    handlerTag={3}
    handlerType="NativeViewGestureHandler"
    innerRef={null}
    onActiveStateChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onPress={[Function]}
    rippleColor="#E4E7EA"
    style={
      [
        {
          "backgroundColor": "#F7F8FA",
          "borderRadius": 4,
          "margin": undefined,
          "marginBottom": 10,
          "marginEnd": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginStart": undefined,
          "marginTop": undefined,
          "marginVertical": undefined,
        },
        {
          "cursor": undefined,
        },
      ]
    }
    underlayColor="#FFFFFF"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderRadius": 4,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessible={true}
      style={{}}
    >
      <View
        accessible={false}
        style={
          {
            "alignItems": "center",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 15,
            "paddingVertical": 14,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#2F343D",
                "fontSize": NaN,
              },
              [
                {
                  "lineHeight": NaN,
                },
                {
                  "left": 15,
                  "position": "absolute",
                },
              ],
              {
                "fontFamily": "custom",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          î¤‰
        </Text>
        <Text
          accessible={false}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Continue with
           
          <Text
            style={
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontWeight": "600",
                "textAlign": "left",
              }
            }
          >
            google
          </Text>
        </Text>
      </View>
    </View>
  </RNGestureHandlerButton>,
  <RNGestureHandlerButton
    activeOpacity={0.5}
    collapsable={false}
    delayLongPress={600}
    handlerTag={4}
    handlerType="NativeViewGestureHandler"
    innerRef={null}
    onActiveStateChange={[Function]}
    onGestureHandlerEvent={[Function]}
    onGestureHandlerStateChange={[Function]}
    onPress={[Function]}
    rippleColor="#E4E7EA"
    style={
      [
        {
          "backgroundColor": "#F7F8FA",
          "borderRadius": 4,
          "margin": undefined,
          "marginBottom": 10,
          "marginEnd": undefined,
          "marginHorizontal": undefined,
          "marginLeft": undefined,
          "marginRight": undefined,
          "marginStart": undefined,
          "marginTop": undefined,
          "marginVertical": undefined,
        },
        {
          "cursor": undefined,
        },
      ]
    }
    underlayColor="#FFFFFF"
  >
    <View
      collapsable={false}
      style={
        {
          "backgroundColor": "#FFFFFF",
          "borderBottomLeftRadius": undefined,
          "borderBottomRightRadius": undefined,
          "borderRadius": 4,
          "borderTopLeftRadius": undefined,
          "borderTopRightRadius": undefined,
          "bottom": 0,
          "left": 0,
          "opacity": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
        }
      }
    />
    <View
      accessible={true}
      style={{}}
    >
      <View
        accessible={false}
        style={
          {
            "alignItems": "center",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingHorizontal": 15,
            "paddingVertical": 14,
            "width": "100%",
          }
        }
      >
        <Text
          allowFontScaling={false}
          selectable={false}
          style={
            [
              {
                "color": "#2F343D",
                "fontSize": NaN,
              },
              [
                {
                  "lineHeight": NaN,
                },
                {
                  "left": 15,
                  "position": "absolute",
                },
              ],
              {
                "fontFamily": "custom",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              {},
            ]
          }
        >
          î¤’
        </Text>
        <Text
          accessible={false}
          style={
            [
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontSize": 16,
                "fontWeight": "400",
                "textAlign": "left",
              },
              {
                "color": "#1F2329",
              },
            ]
          }
        >
          Continue with
           
          <Text
            style={
              {
                "backgroundColor": "transparent",
                "fontFamily": "Inter",
                "fontWeight": "600",
                "textAlign": "left",
              }
            }
          >
            apple
          </Text>
        </Text>
      </View>
    </View>
  </RNGestureHandlerButton>,
]
`;
